### PR TITLE
Update troubleshooting guide

### PR DESCRIPTION
**Pull request by scalar-boney**
Update troubleshooting guide

* `terraform refresh` shows the following error
```
module.cassandra.azurerm_virtual_machine_data_disk_attachment.cassandra_data_volume_attachment[0]: Refreshing state... [id=/subscriptions/b647bf0e-0566-4b28-ac8c-37b24f3b7061/resourceGroups/psim-azure-hvidlnk/providers/Microsoft.Compute/virtualMachines/cassandra-1/dataDisks/data-disk-cassandra0]
module.cassandra.azurerm_virtual_machine_data_disk_attachment.cassandra_data_volume_attachment[2]: Refreshing state... [id=/subscriptions/b647bf0e-0566-4b28-ac8c-37b24f3b7061/resourceGroups/psim-azure-hvidlnk/providers/Microsoft.Compute/virtualMachines/cassandra-3/dataDisks/data-disk-cassandra2]
module.cassandra.azurerm_virtual_machine_data_disk_attachment.cassandra_data_volume_attachment[1]: Refreshing state... [id=/subscriptions/b647bf0e-0566-4b28-ac8c-37b24f3b7061/resourceGroups/psim-azure-hvidlnk/providers/Microsoft.Compute/virtualMachines/cassandra-2/dataDisks/data-disk-cassandra1]
module.cassandra.module.cassandra_provision.null_resource.cassandra[2]: Refreshing state... [id=196610524335755786]
module.cassandra.module.cassandra_provision.null_resource.cassandra[0]: Refreshing state... [id=7697572524007658753]
module.cassandra.module.cassandra_provision.null_resource.cassandra[1]: Refreshing state... [id=3515917159123694899]

Error: Virtual Machine "cassandra-1" (Resource Group "psim-azure-hvidlnk") was not found


Releasing state lock. This may take a few moments...
```

* Node with data disk recovery also failed while using the following commands
```
terraform taint "module.cassandra.module.cassandra_cluster.azurerm_virtual_machine.vm-linux[0]"
terraform taint "module.cassandra.azurerm_managed_disk.cassandra_data_volume[0]"

terraform apply
```

### CODE REVIEW

File: docs/TroubleshootingGuide.md [hunk 1]
```diff
@@ -17,12 +17,28 @@ If you accidentally delete a node that contains an additional data disk in Azure
...
-* Do `terraform refresh` or `terraform state rm` as follows.
+* Do `terraform state rm` as follows.
 ```console
 terraform state rm "module.cassandra.module.cassandra_cluster.azurerm_virtual_machine.vm-linux[0]"
 terraform state rm "module.cassandra.azurerm_virtual_machine_data_disk_attachment.cassandra_data_volume_attachment[0]"
 ```
 * Do `terraform apply`
 
-Follow [Cassandra Post Recovery Steps](CassandraOperation.md#post-recovery-steps)
+Follow [Cassandra Post Recovery Steps](CassandraOperation.md#post-terraform-steps)
```
Comment (tei-k) [inline docs/TroubleshootingGuide.md:27]:
```suggestion
Follow [Cassandra Post Terraform Steps](CassandraOperation.md#post-terraform-steps)
```

File: docs/TroubleshootingGuide.md [hunk 2]
```diff
@@ -17,12 +17,28 @@ If you accidentally delete a node that contains an additional data disk in Azure
...
+* Delete the os-disk If the node is terminated.
+* Do `terraform state rm` as follows.
+```console
+terraform state rm "module.cassandra.module.cassandra_cluster.azurerm_virtual_machine.vm-linux[0]"
+terraform state rm "module.cassandra.azurerm_virtual_machine_data_disk_attachment.cassandra_data_volume_attachment[0]"
+terraform state rm "module.cassandra.azurerm_managed_disk.cassandra_data_volume[0]"
+```
+* Do `terraform apply`
+
+Follow [Cassandra Post Recovery Steps](CassandraOperation.md#post-terraform-steps-2)
```
Comment (tei-k) [inline docs/TroubleshootingGuide.md:42]:
```suggestion
Follow [Cassandra Post Terraform Steps](CassandraOperation.md#post-terraform-steps-2)
```

File: docs/TroubleshootingGuide.md (hunk 1 above)
Comment (scalar-boney) [reply docs/TroubleshootingGuide.md]: Thank you for your suggestion.
Document updated accordingly.

File: docs/TroubleshootingGuide.md (hunk 2 above)
Comment (scalar-boney) [reply docs/TroubleshootingGuide.md]: Thank you for your suggestion.
Document updated accordingly.

**Review by tei-k — APPROVED** [review]
LGTM!

File: docs/TroubleshootingGuide.md [hunk 3]
```diff
@@ -17,12 +17,28 @@ If you accidentally delete a node that contains an additional data disk in Azure
...
 ```console
 terraform state rm "module.cassandra.module.cassandra_cluster.azurerm_virtual_machine.vm-linux[0]"
 terraform state rm "module.cassandra.azurerm_virtual_machine_data_disk_attachment.cassandra_data_volume_attachment[0]"
 ```
 * Do `terraform apply`
 
-Follow [Cassandra Post Recovery Steps](CassandraOperation.md#post-recovery-steps)
+Follow [Cassandra Post Terraform Steps](CassandraOperation.md#post-terraform-steps)
+
+### Recover a node with data disk from accidental deletion of a node and data disk in Azure
```
Comment (feeblefakie) [inline docs/TroubleshootingGuide.md:29]: Is this step only required in Azure? (no action needed for AWS?)

Comment (feeblefakie) [inline docs/TroubleshootingGuide.md:29]: Also, are you talking about a case where both the node and the data disk are deleted?
If so, the sentence doesn't sound like it.
```suggestion
### Recover a node with a data disk from accidental deletion of both the node and the data disk in Azure
```

File: docs/TroubleshootingGuide.md [hunk 4]
```diff
@@ -17,12 +17,28 @@ If you accidentally delete a node that contains an additional data disk in Azure
...
+If you accidentally delete a node and data disk in Azure, you can recover that node and data disk using the following steps.
+
+Please try the following
```
Comment (feeblefakie) [inline docs/TroubleshootingGuide.md:32]: You've already said `using the following steps` in the above sentence. Isn't it redundant?
(If you just copied without thinking, please don't do that)

**Review by feeblefakie — CHANGES_REQUESTED** [review]
Left some comments.

File: docs/TroubleshootingGuide.md (hunk 3 above)
Comment (scalar-boney) [reply docs/TroubleshootingGuide.md:29]: > Is this step only required in Azure? (no action needed for AWS?)

Yes, It is only required for Azure

Comment (scalar-boney) [reply docs/TroubleshootingGuide.md:29]: > Also, are you talking about a case where both the node and the data disk are deleted?

Yes,
Above suggestion updated in the document. 
Thank you for the suggestion.

File: docs/TroubleshootingGuide.md (hunk 4 above)
Comment (scalar-boney) [reply docs/TroubleshootingGuide.md:32]: > You've already said using the following steps in the above sentence. Isn't it redundant?

Sorry, It's my mistake.
 I have removed `Please try the following` from the document.

**Review by feeblefakie — APPROVED** [review]
LGTM

**Review by tei-k — APPROVED** [review]

**Pull request merged**